Source/Particles/Collision/MCCScattering.H

```diff
@@ -0,0 +1,304 @@
+/* Copyright 2021 Roelof Groenewald
+ *
+ * This file is part of WarpX.
+ *
+ * License: ????
+ */
+#ifndef WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+#define WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+
+#include "MCCProcess.H"
+#include "Utils/WarpXConst.H"
+#include <AMReX_Random.H>
+
+/** @file
+ *
+ * This file contains the implementation of the scattering processes available
+ * in the MCC handling.
+ */
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void getRandomVector ( amrex::Real& x, amrex::Real& y, amrex::Real& z,
+                       amrex::RandomEngine const& engine )
+{
+  using std::sqrt;
+  using std::cos;
+  using std::sin;
```

Comment:
I just noticed - the indentation is inconsistent here. That was probably my fault when I introduced these lines. I'll correct it momentarily